### PR TITLE
Documentation: Update documentation to refer only to Docker Compose v2 command

### LINKED ISSUE #4649: [BUG] docker-compose v1 issues
### Description

When running the install script as specified in https://paperless.readthedocs.io/en/latest/setup.html via docker-compose run --rm webserver createsuperuser, the script fails iff PAPERLESS_OCR_ROTATE_PAGES is set to true in the docker-compose.yml.

### Steps to reproduce

1. Create fresh installation via docker compose file, Don't specify admin username or password in the compose file. Specify PAPERLESS_OCR_ROTATE_PAGES: true, as described in https://docs.paperless-ngx.com/configuration/. (I did PAPERLESS_OCR_ROTATE_PAGES_THRESHOLD 8)
2. pull and up -d, wait. Start will be successful and web page will be responding.
3. try to create the user via docker-compose run --rm webserver createsuperuser
4. ERROR: The Compose file './docker-compose.yml' is invalid because:
services.webserver.environment.PAPERLESS_OCR_ROTATE_PAGES contains true, which is an invalid type, it should be a string, number, or a null

### Webserver logs

```bash
no relevant entry.
```

### Browser logs

_No response_

### Paperless-ngx version

1.17.4

### Host OS

Debian 6.1.52-1 (2023-09-07) x86_64 GNU/Linux

### Installation method

Docker - official image

### Browser

_No response_

### Configuration changes

described above

### Other

_No response_

### PR DESCRIPTION
**Pull request by stumpylog**
## Proposed change

Updates `docker-compose` to refer to either `docker compose` (the command for v2) or Docker Compose (the general tool, not the command).  v1 is no longer supported by Docker since July 2023 and we shouldn't refer to it anymore either.

Fixes #4649 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain):

## Checklist:

- [ ] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/setup.md [hunk 1]
```diff
@@ -62,19 +65,19 @@ steps described in [Docker setup](#docker_hub) automatically.
...
 
-3.  Install [Docker](https://www.docker.com/) and
-    [docker-compose](https://docs.docker.com/compose/install/).
+3.  Install [Docker](https://docs.docker.com/engine/install/) and
+    [Docker Compose](https://docs.docker.com/compose/install/).
 
     !!! warning
 
         If you want to use the included `docker-compose.*.yml` file, you
         need to have at least Docker version **17.09.0** and docker-compose
```
Comment (shamoon) [inline docs/setup.md:74]:
```suggestion
        need to have at least Docker version **17.09.0** and Docker Compose
```

**Review by shamoon — COMMENTED** [review]
Noticed 1 other spot. I agree with the docs change but is there a real downside to leaving the script as is? I realize it’d be the same thing where a user could say “script let me use docker-compose but docs don’t say that” but I think that’s ok because people still using -compose mostly realize they’re behind

**Comment by shamoon** [comment]
And thanks :)

**Comment by stumpylog** [comment]
I think it is worth updating the installation script.  v2 is easy to install and given it acts differently (arguably better) for setting certain environment variables, I don't see a reason not to require it at this point.  Perhaps they don't realize they are behind and will raise something more difficult because they're using v1 instead of v2

**Review by shamoon — APPROVED** [review]
Ok, I’m good with that rationale 👍

**Pull request merged**